source/ncengine/graphics2/NcGraphicsImpl2.cpp

```diff
@@ -247,7 +251,9 @@ void NcGraphicsImpl2::Run()
     context.ClearRenderTarget(pRTV, &ClearColor.x, Diligent::RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
     context.ClearDepthStencil(pDSV, Diligent::CLEAR_DEPTH_FLAG, 1.f, 0, Diligent::RESOURCE_STATE_TRANSITION_MODE_TRANSITION);
 
+    m_shaderBindings.Update(renderState, context);
```

Comment:
I'm not totally sure how we want to handle getting render state to its destination(s). There is very little happening now, but maybe this needs to get more sophisticated.

Reply:
This is great - provided we can guarantee we only map constant buffers here and don't require an update of a descriptor set (which happens when you change the size of an array). See my other comment on the layer between AssetDispatch and the GPU for more context.